### PR TITLE
Refactor function tickets assignment logic in ExpressionFuzzer

### PR DESCRIPTION
**Pull request by kagamiori**
Summary:
Refactor the logic in ExpressionFuzzer of assigning weights to functions according to 
specified tickets. This is needed for adding support of dereference in the next diff.

Differential Revision: D44081699

### CODE REVIEW

File: velox/expression/tests/ExpressionFuzzer.cpp [hunk 1]
```diff
@@ -502,15 +465,8 @@ ExpressionFuzzer::ExpressionFuzzer(
...
-  auto functionsToTickets = getTicketsForFunctions();
-  auto getTickets = [&functionsToTickets](const std::string& funcName) {
-    auto itr = functionsToTickets.find(funcName);
-    int tickets = 1;
-    if (itr != functionsToTickets.end()) {
-      tickets = itr->second;
-    }
-    return tickets;
-  };
+  getTicketsForFunctions();
```
Comment (mbasmanova) [inline velox/expression/tests/ExpressionFuzzer.cpp:468]: These are nice refactoring, but it makes it hard to figure out the actual changes. Would it be possible to extract refactoring into a separate PR?

**Review by mbasmanova — COMMENTED** [review]
@kagamiori Wei, thank you for adding support for row_constructor and dereference to the Fuzzer. Looks good overall. Curious if you found any bugs with these changes.

@bikramSingh91 Bikram, would you help review this PR?

File: velox/expression/FieldReference.cpp [hunk 2]
```diff
@@ -103,9 +103,29 @@ void FieldReference::evalSpecialFormSimplified(
          return static_cast<Expr*>(expr)->toString();
        },
        this});
-
-  auto row = context.row();
-  result = row->childAt(index(context));
-  BaseVector::flattenVector(result, rows.end());
+  VectorPtr input;
```
[the remark this answers is not in the document]
Comment (kagamiori) [reply velox/expression/FieldReference.cpp]: Yes, this has been landed in a separate PR https://github.com/facebookincubator/velox/pull/4354.

File: velox/expression/tests/ExpressionFuzzer.cpp (hunk 1 above)
Comment (kagamiori) [reply velox/expression/tests/ExpressionFuzzer.cpp:464]: I split the refactoring from the change of dereference support. Thanks for the suggestion!
